src/redux/books/books.js

```diff
@@ -1,37 +1,81 @@
-const ADD_BOOK = 'ADD_BOOK';
-const REMOVE_BOOK = 'REMOVE_BOOK';
-const books = [
-  { id: 1, author: 'Book-1 author', title: 'Book-1' },
-  { id: 2, author: 'Book-2 author', title: 'Book-2' },
-  { id: 3, author: 'Book-3 author', title: 'Book-3' },
-  { id: 4, author: 'Book-4 author ', title: 'Book-4' },
-];
-
-export const AddBookFunction = (NewBook) => ({
+const ADD_BOOK = 'bookStore/books/ADD_BOOK';
+const REMOVE_BOOK = 'bookStore/books/REMOVE_BOOK';
+const FETCH_BOOKS = 'bookstore/books/FETCH_BOOKS';
+const urlLink = 'https://us-central1-bookstore-api-e63c8.cloudfunctions.net/bookstoreApi/apps/IEXNcgPY8gfuqg4iq5ZV/books';
+
+const initialState = [];
+
+export const addBook = (payload) => ({
   type: ADD_BOOK,
-  payload: NewBook,
+  payload,
 });
-export const RemoveBookFunction = (id) => ({
+
+export const removeBook = (payload) => ({
   type: REMOVE_BOOK,
-  id,
+  payload,
 });
 
-const BooksReducer = (state = books, action) => {
+export const fetchBook = (payload) => ({
+  type: FETCH_BOOKS,
+  payload,
+});
+
+const BookReducer = (state = initialState, action) => {
   switch (action.type) {
     case ADD_BOOK:
-      return [
-        ...state,
-        {
-          id: Date.now(),
-          title: action.payload.title,
-          author: action.payload.author,
-        },
-      ];
+      return [...state, action.payload];
+
     case REMOVE_BOOK:
-      return state.filter((book) => book.id !== action.id);
+      return state.filter((book) => book.id !== action.payload);
+
+    case FETCH_BOOKS:
+      return action.payload;
+
     default:
       return state;
   }
 };
 
-export default BooksReducer;
+export const fetchBooksList = () => async (dispatch) => {
+  const booksList = await fetch(`${urlLink}`).then((response) => response.json());
+  const booksID = Object.keys(booksList);
+  const formBooks = [];
+  booksID.map((key) => formBooks.push({
+    id: key,
+    title: booksList[key][0].title,
+    author: booksList[key][0].author,
+    category: booksList[key][0].category,
+  }));
+  dispatch(fetchBook(formBooks));
+};
+
+export const postBook = (newBook) => async (dispatch) => {
+  await fetch(`${urlLink}`, {
+    method: 'POST',
+    body: JSON.stringify({
+      item_id: newBook.id,
+      title: newBook.title,
+      author: newBook.author,
+      category: newBook.category,
+    }),
+    headers: {
+      'content-type': 'application/json; charset=UTF-8',
+    },
+  });
+  dispatch(addBook(newBook));
+};
+
+export const deleteBook = (id) => async (dispatch) => {
+  await fetch(`${urlLink}/${id}`, {
+    method: 'DELETE',
+    body: JSON.stringify({
+      item_id: id,
+    }),
+    headers: {
+      'content-type': 'application/json; charset=UTF-8',
+    },
+  });
+  dispatch(removeBook(id));
+};
```

Comment:
- [ ] As requested of you by the previous reviewer,  Kindly note that it is a requirement that you make use of [createasyncthunk](https://redux-toolkit.js.org/api/createAsyncThunk) when making your API calls according to the project requirements.